includes/gateways/paypal/includes/class-wc-gateway-paypal-ipn-handler.php

```diff
@@ -339,6 +339,16 @@ protected function send_ipn_email_notification( $subject, $message ) {
 		$mailer             = WC()->mailer();
 		$message            = $mailer->wrap_message( $subject, $message );
 
+		$ipn_notification = 'yes';
+		$woocommerce_paypal_settings = get_option('woocommerce_paypal_settings');
+		if ( isset($woocommerce_paypal_settings['ipn_notification']) && $woocommerce_paypal_settings['ipn_notification'] != '' ) {
+		    $ipn_notification = $woocommerce_paypal_settings['ipn_notification'];
+		}
+
+		if ( $ipn_notification == 'no' ) {
+			return;
+		}
+
```

Comment:
This works, but I think it can be improved. What about something like

```
$woocommerce_paypal_settings = get_option('woocommerce_paypal_settings');
if ( ! empty( $woocommerce_paypal_settings['ipn_notification'] ) && 'no' === $woocommerce_paypal_settings['ipn_notification'] ) {
    return;
}
```

Much shorter and cleaner.

It won't be needed if you go with the code I just provided, but there are some coding style things around [spacing](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#space-usage) and [Yoda conditions](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions) that need to be fixed in this section of code.